app/src/main/java/com/example/shiv/mdbpokedexproject2/MainActivity.java

```diff
@@ -0,0 +1,282 @@
+                    return true;
+
+                }
+
+            case R.id.search:
+
+                return true;
+
+            case R.id.filter:
+
+                Intent intent = new Intent(getApplicationContext(), FilterActivity.class);
+                startActivity(intent);
+
+                return true;
+
+            case R.id.randomPoke:
+                ArrayList <Pokedex.Pokemon> pokedex = new Pokedex().getPokemon();
+                Random integer = new Random();
+                ArrayList <Pokedex.Pokemon> randPoke = new ArrayList<>();
+                for (int i = 0; i < 20; i++) {
+                    randPoke.add(pokedex.get(integer.nextInt(pokedex.size())));
+                }
+                mModels = randPoke;
+                pokedexAdapter.pokemons = randPoke;
+                rview.setAdapter(pokedexAdapter);
+
+                this.rview.scrollToPosition(0);
+                Toast toast3 = Toast.makeText(getApplicationContext(), "Enjoy your 20 random Pokemon!", Toast.LENGTH_SHORT);
+                toast3.show();
+
+
+
+
+            default:
+                // If we got here, the user's action was not recognized.
+                // Invoke the superclass to handle it.
+                return super.onOptionsItemSelected(item);
+
+        }
+    }
+
+    //filter the pokemon based on min_attack, min_defense, and min_health
+    private static ArrayList<Pokedex.Pokemon> filterAttributes(List<Pokedex.Pokemon> models) {
+
+        final ArrayList<Pokedex.Pokemon> filteredModelList = new ArrayList<>();
+        for (Pokedex.Pokemon model : models) {
+
+            if (Integer.parseInt(model.attack) >= MainActivity.min_attack && Integer.parseInt(model.defense) >= MainActivity.min_defense && Integer.parseInt(model.hp) >= MainActivity.min_health) {
+                if (displayAllTypes) {
+                    filteredModelList.add(model);
+                }
+                else {
+                    if (inList(MainActivity.type,model.type)) {
+                        filteredModelList.add(model);
+                    }
+                }
+            }
+        }
+        return filteredModelList;
+    }
+
+
+
+    private static boolean inList(ArrayList<String> list, String str) {
+        boolean in = false;
+        for (int i = 0; i < list.size(); i++) {
+            if (list.get(i).equals(str)) {
+                in = true;
+            }
+        }
+        return in;
+    }
+
+    @Override
+    protected void onResume() {
+        super.onResume();
+
+        //Filter the pokemon based on the attributes set in FilterActivity
+        final ArrayList<Pokedex.Pokemon> filteredModelList = filterAttributes(mModels);
+        mModels = filteredModelList;
+        pokedexAdapter.pokemons = filteredModelList;
+
+        this.rview.scrollToPosition(0);
+    }
+
+    //Lost points on this last time.
+    //Is there any reason to include onPause method if it just invokes super?
+    @Override
```

Comment:
lol last time was because there was something to do in onPause (pause the timer) this time nothing is really happening